extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -233,6 +290,14 @@ oppia.directive('imageWithRegionsEditor', [
           }
           $scope.userIsCurrentlyDrawing = false;
           $scope.userIsCurrentlyDragging = false;
+          $scope.userIsCurrentlyResizing = false;
+          if ($scope.outOfRegion) {
+            $scope.xDirection = 0;
+            $scope.yDirection = 0;
+          }
```

Comment:
One work around that I was thinking was we can subtract the margin i.e, the y and the position where the cursor is clicked and subtract this whenever the direction changes so that the cursor does not get outside the region. But it will complicate it a bit. Should I go ahead with it?